tests/configuration_tests.py

```diff
@@ -0,0 +1,119 @@
+                'test event 1': [
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="foobar", arg="", model="auth.User"),
+                    dict(name="", arg="invalid arg", model="auth.User"),
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                ],
+                'test event 2': [
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                    dict(foobar="abc"),
+                ],
+                'test event 3': [
+                    dict(name="User1", arg="user1", model="invalid_app.User"),
+                    dict(name="User2", arg="user2", model="auth.InvalidModel"),
+                ],
+            },
+        }
+
+        util.validate_configuration(config)
+
+        expected_messages = set([
+            "invalid event name: ''",
+            "reused data name for event 'test event 1': 'Reused Name'",
+            "reused data argument for event 'test event 1': 'reused_arg'",
+            "invalid data argument for event 'test event 1': ''",
+            "invalid data argument for event 'test event 1': 'invalid arg'",
+            "invalid data name for event 'test event 1': ''",
+            "unexpected data key for event 'test event 2': 'foobar'",
+            "missing data key for event 'test event 2': 'name'",
+            "missing data key for event 'test event 2': 'arg'",
+            "missing data key for event 'test event 2': 'model'",
+            ("bad model 'invalid_app.User' for event 'test event 3': "
+             "No installed app with label 'invalid_app'."),
+            ("bad model 'auth.InvalidModel' for event 'test event 3': "
+             "App 'auth' doesn't have a 'invalidmodel' model."),
+        ])
+        actual_messages = [message.getMessage()
+                           for message in handler.messages]
+
+        for message in actual_messages:
+            if message not in expected_messages:
+                raise self.failureException("got unexpected log message '{}'"
+                                            .format(message))
+
+            expected_messages.remove(message)
+
+        if expected_messages:
+            raise self.failureException(
+                "expected log message(s) missing: {}"
+                .format(list(expected_messages)))
+
+    @override_settings()
+    @mock.patch('django_emarsys.util.validate_configuration')
+    def test_validate_settings(self, mock_validate_configuration):
+        expected_config = {key: getattr(settings, key)
+                           for key in dir(settings)
+                           if hasattr(settings, key)}
+        util.validate_settings()
+
+        mock_validate_configuration.assert_called_with(expected_config)
```

Comment:
Ok, I'll try to understand your arguments:

"it's not entirely trivial":

What is?

"I want to make sure that validate_configuration is called correctly"

I'm 100% sure that that's the case by just looking at the code. That part is outright trivial.

"It also makes sure `validate_settings()` is called at least once in tests."

If this is about coverage: coverage has diminishing returns. To achive the last 30 or 20 or 5%, you end up testing trivial code, where you expend an effort to keep the test happy that's unproportionally higher than the probability that the test will catch a bug.

"in that case it's worth a second look at the change"

Do you mean that you have to make the change in two places, and that's the second look? If so, the second look is provided by a code review.

I agree time running time is insignificant in this particular case.

"maintenance is a non-issue"

How so?

(Feel free to postpone answering to a later date.)